src/tap/src/tapcell.cpp

```diff
@@ -57,17 +63,1091 @@ Tapcell::~Tapcell()
 {
 }
 
-void
-Tapcell::init(Tcl_Interp *tcl_interp,
-	   odb::dbDatabase *db)
+void Tapcell::init(odb::dbDatabase *db, utl::Logger* logger)
 {
   db_ = db;
+  logger_ = logger;
+}
+
+void Tapcell::reset()
+{
+  Tapcell::phy_idx_ = 0;
+  Tapcell::filled_sites_.clear();
+}
+
+//}
+
+//---------------------------------------------------------------
+
+void Tapcell::clear (const char* tap_prefix, const char* endcap_prefix) {
```

Comment:
I think you can make `tap_prefix` and `endcap_prefix` into class members, so you don't have to pass them as arguments in multiple functions.